exec/java-exec/src/main/java/org/apache/drill/exec/ops/QueryContext.java

```diff
@@ -77,6 +78,7 @@
   private final SchemaTreeProvider schemaTreeProvider;
   /** Stores constants and their holders by type */
   private final Map<String, Map<MinorType, ValueHolder>> constantValueHolderCache;
+  private StatementType stmtType = StatementType.UNKNOWN;
```

Comment:
Maybe initialize to null (the default) and require that the statement type be set later. Or, pass the statement type into the constructor and make the type a `final` variable showing that the statement type cannot change over time.